osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -534,5 +523,25 @@ private void updateUserPlayingState(int userId, MultiplayerUserState state)
             else
                 CurrentMatchPlayingUserIds.Remove(userId);
         }
+
+        private Task scheduleAsync(Action action)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            Scheduler.Add(() =>
+            {
+                try
+                {
+                    action();
+                    tcs.SetResult(true);
+                }
+                catch (Exception ex)
+                {
+                    tcs.SetException(ex);
+                }
+            });
```

Comment:
Is there a reason to not permit this to potentially run inline like all the other operations around can?